integration_tests/__tests__/only_changed.test.js

```diff
@@ -73,3 +73,60 @@ test('run only changed files', () => {
   expect(stderr).toMatch('PASS  __tests__/file2.test.js');
   expect(stderr).toMatch('PASS  __tests__/file3.test.js');
 });
+
+test('onlyChanged in config is overwritten by --all', () => {
+  writeFiles(DIR, {
+    '.watchmanconfig': '',
+    '__tests__/file1.test.js': `require('../file1'); test('file1', () => {});`,
+    'file1.js': 'module.exports = {}',
+    'package.json': JSON.stringify({jest: {onlyChanged: true}}),
+  });
+  let stderr;
+  let stdout;
+
+  ({stdout} = runJest(DIR));
+  expect(stdout).toMatch(/Jest can only find uncommitted changed files/);
+
+  run(`${GIT} init`, DIR);
+  run(`${GIT} add .`, DIR);
+  run(`${GIT} commit -m "first"`, DIR);
+
+  ({stdout} = runJest(DIR));
+  expect(stdout).toMatch('No tests found related to files');
+
+  ({stderr} = runJest(DIR, ['--lastCommit']));
+  expect(stderr).toMatch('PASS  __tests__/file1.test.js');
+
+  writeFiles(DIR, {
+    '__tests__/file2.test.js': `require('../file2'); test('file2', () => {});`,
+    '__tests__/file3.test.js': `require('../file3'); test('file3', () => {});`,
+    'file2.js': 'module.exports = {}',
+    'file3.js': `require('./file2')`,
+  });
+
+  ({stderr} = runJest(DIR));
+
+  expect(stderr).not.toMatch('PASS  __tests__/file1.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file2.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file3.test.js');
+
+  run(`${GIT} add .`, DIR);
+  run(`${GIT} commit -m "second"`, DIR);
+
+  ({stderr} = runJest(DIR));
+  expect(stdout).toMatch('No tests found related to files');
+
+  writeFiles(DIR, {
+    'file2.js': 'module.exports = {modified: true}',
+  });
+
+  ({stderr} = runJest(DIR));
+  expect(stderr).not.toMatch('PASS  __tests__/file1.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file2.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file3.test.js');
+
+  ({stderr} = runJest(DIR, ['--all']));
+  expect(stderr).toMatch('PASS  __tests__/file1.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file2.test.js');
+  expect(stderr).toMatch('PASS  __tests__/file3.test.js');
```

Comment:
yargs has `conflicts` option that i tried using for this, but then i though that it might be ok.
i think a use case of aliasing `jest` to something like `jest --someOptions --onlyChanged`
and then running it with `jest --all` is pretty common and `--all` is explicitly said to overwrite `-o`
but idk. @cpojer what do you think?